### PR TITLE
Fix filtering of non-api stuff in DocletMultiplexer

### PR DESCRIPTION
**Pull request by anistor**
* fix DocletMultiplexer; make it filter elements based on ```@private/@public``` javadoc tags
* remove java 8 support as it was no longer functional anyway

Original filtering algo is here: https://github.com/infinispan/infinispan/blob/9.3.7.Final/javadoc/doclets/src/main/java/org/infinispan/doclets/api/PublicAPIFilterHandler.java#L45

This implementation blindly copies that. Might be good or bad. I suspect we should check the outers recursively. I wonder how that heck did that old code work.

### CODE REVIEW

**Comment by anistor** [comment]
Updated

**Comment by anistor** [comment]
From these 3 commits only "Exclusion of @Private elements from javadoc " is actually relevant

**Comment by anistor** [comment]
Seems this is not picking up public classes from packages that are not explicitly marked public. Need to see what can be done for that.

File: java11/pom.xml [hunk 1]
```diff
@@ -29,12 +39,38 @@
                <release>11</release>
             </configuration>
          </plugin>
+
+         <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
```
Comment (tristantarrant) [inline java11/pom.xml:44]: Why are you shading the dependency ?

Comment (anistor) [reply java11/pom.xml:44]: I don't want bytebuddy to escape. It's toxic.

**Comment by anistor** [comment]
I've just updated the pr.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**